### PR TITLE
Story/vspc 86

### PR DESCRIPTION
**Pull request by prjadhav14**
This development involves adding indicators for space links in staff screens and in the linked space dropdown when user creates a new link or edits existing link.

### CODE REVIEW

File: vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp [hunk 1]
```diff
@@ -64,13 +64,22 @@ $( document ).ready(function() {
...
+		unpublishedSpaceElement.css('font-size', "12px");
+		
 		//append to form element that you want .
 		$("#space").append(inputRotation);
 		$("#space").append(inputLabel);
 		$("#space").append(inputType);
 		$("#space").append(inputLinkedSpace);
 		$("#space").append(link);
+		$("#space").append(unpublishedSpaceElement);
+		//$(".spaceLink-${link.link.id}").append(unpublishedSpaceElement);
```
Comment (coder-aakash994) [inline vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp:82]: remove comment lines

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
what aakash said. afterwards should be good.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
there is actually an issue I missed before. For adding a new space link, the drop down is wider than the modal:
image

**Comment by jdamerow** [comment]
If I create a new space link and choose an unpublished space, I have to reload the page to get the alert icon.

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceLinkController.java [hunk 2]
```diff
@@ -77,6 +77,7 @@
         linkNode.put("displayId", display.getId());
         linkNode.put("x", display.getPositionX());
         linkNode.put("y", display.getPositionY());
+        linkNode.put("likedSpaceStatus", spaceManager.getSpace(linkedSpaceId).getSpaceStatus().toString());
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceLinkController.java:80]: do you mean "linkedSpaceStatus"?

File: vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp [hunk 3]
```diff
@@ -777,6 +792,7 @@ $( document ).ready(function() {
 		}
 
 		$("#space").append(link);
+		$("#space").append(unpublishedSpaceElement);
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp:795]: if `show` is false, then `unpublishedSpaceElement` is undefined here, isn't it?

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- When I delete a link to an unpublished space, the warning triangle stays.
- when I edit a space link from an unpublished to a published space, the warning triangle stays.

**Comment by jdamerow** [comment]
Also, it seems like when I have a link to an unpublished space, I can't delete any other link anymore. And there are other oddities but not sure if that's related to the changes in this ticket or generally a problem (https://diging.atlassian.net/browse/VSPC-119).

**Comment by jdamerow** [comment]
there are still issues.
- I create a link to an unpublished space
- I click the link to edit it
- i select a published space
- i click save, nothing happens
- when I reload the page, the changes have been persisted (the link links to a published space), but something seems to go wrong after the save has been executed (before the modal closes)

It works the other way though, I can choose an unpublished space and that works fine.

**Comment by prjadhav14** [comment]
> there are still issues.
> 
>     * I create a link to an unpublished space
> 
>     * I click the link to edit it
> 
>     * i select a published space
> 
>     * i click save, nothing happens
> 
>     * when I reload the page, the changes have been persisted (the link links to a published space), but something seems to go wrong after the save has been executed (before the modal closes)
> 
> 
> It works the other way though, I can choose an unpublished space and that works fine.

is working fine in local.

**Comment by jdamerow** [comment]
This seems to be the problem:
```
</pre><p><b>Root Cause</b></p><pre>java.lang.NullPointerException
	edu.asu.diging.vspace.web.staff.EditSpaceLinkController.editSpaceLink(EditSpaceLinkController.java:53)
```

**Comment by jdamerow** [comment]
still throwing a nullpointer:
```
</pre><p><b>Root Cause</b></p><pre>java.lang.NullPointerException
	edu.asu.diging.vspace.web.staff.EditSpaceLinkController.editSpaceLink(EditSpaceLinkController.java:57)
```

File: vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp [hunk 4]
```diff
@@ -849,6 +868,25 @@ $( document ).ready(function() {
...
+			    unpublishedSpaceElement=$('<i data-link-id = "unpublished-' + spaceLink["id"] + '" class="fa fa-exclamation-triangle fa-lg unpublishedSpaceClass unpublishedClass-' + spaceLink["id"] + '" style="color: #bfb168;"></i>')
+				unpublishedSpaceElement.css('position', 'absolute');
+				unpublishedSpaceElement.css('left', parseInt(spaceLink["x"]) + posX + 25);
+				unpublishedSpaceElement.css('top', parseInt(spaceLink["y"]) + posY - 13);
+				unpublishedSpaceElement.css('transform', 'rotate(' +spaceLink["rotation"]+ 'deg)');
+				unpublishedSpaceElement.css('font-size', "12px");
+				$("#space").append(unpublishedSpaceElement);
+			}else{
+			    $('[data-link-id="unpublished-' + spaceLink["id"] + '"]').remove();
+			}
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp:884]: indentation

**Review by jdamerow — CHANGES_REQUESTED** [review]
just those few indentation issues and then it should be good to go

**Pull request merged**